uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/absence-tab/components/absence-tab-entitlements.js

```diff
@@ -10,13 +14,84 @@ define([
       return settings.pathTpl + 'components/absence-tab-entitlements.html';
     }],
     controllerAs: 'entitlements',
-    controller: ['$log', controller]
+    controller: ['$controller', '$log', '$q', '$rootScope', 'AbsenceType', 'Contract', controller]
   });
 
-  function controller($log) {
+  function controller ($controller, $log, $q, $rootScope, AbsenceType, Contract) {
     $log.debug('Component: absence-tab-entitlements');
 
-    var vm = {};
+    var vm = this;
+
+    vm.contactId = vm.contactId || CRM.vars.leaveAndAbsences.contactId;
```

Comment:
1. At this moment this is PR is becoming too big and its tough to review so many stuff, originally this ticket was meant for only component creation. But as you found out that you need to also create API, Modal and Instance for Contract, a new ticket should have been created to create those first and then move on to this ticket where the component will be developed. I still suggest to create a new ticket so that PCHR-2007 only contains the API, Modal and Instance . And as part of the new ticket the component will be developed.

2. This implementation is not correct, As you can see from the ticket description, it was mentioned that a component named `<contract-entitlements contact-id="contactID"></contract-entitlements>` should be created. 
Because  `absence-tab-entitlements` is the container component for this tab, but as it has 2 child components i.e. Contract and Annual, we would need to create 2 child components.
The ticket description should always be followed while implementing the same, or if you think there is a better solution it needs to be discussed on the room.

For the above mentioned reasons I am not reviewing the component part of this PR. Please feel free to discuss in the room.
cc @AkA84